### ISSUE TITLE
Saving specific tag corrupts tag (`SizeMismatch`)

### ISSUE THREAD

**Issue by probablykasper** [Serial-ATA/lofty-rs#469]
### Reproducer

```rs
use lofty::config::WriteOptions;
use lofty::file::TaggedFileExt;
use lofty::read_from_path;
use lofty::tag::{Accessor, TagExt};

fn main() {
	let bugged_song_path = "bugged.opus";
	std::fs::copy("song.opus", bugged_song_path).unwrap();

	let mut tagged_file = read_from_path(bugged_song_path).unwrap();
	let tag = tagged_file.primary_tag_mut().unwrap();

	tag.set_artist("Kuraine".to_owned());

	tag.save_to_path(bugged_song_path, WriteOptions::default())
		.unwrap();

	// Open again
	match read_from_path(bugged_song_path) {
		Ok(_) => {}
		Err(e) => println!("{:?}", e),
	};
}
```

### Summary

Saving this specific combination of metadata corrupts the tag. Opening the corrupted file's tag gives a `SizeMismatch` error

### Assets

[song.opus.zip](https://github.com/user-attachments/files/17449274/song.opus.zip)

Corrupt file created by reproducer:
[bugged.opus.zip](https://github.com/user-attachments/files/17449275/bugged.opus.zip)

**Comment by Serial-ATA** [comment]
Thanks, that comes from a case I forgot to test when I fixed #350. You're right that it is this tag specifically. The packet wasn't being properly terminated when the last page size is divisible by 255. I'll get it fixed soon.

**Comment by probablykasper** [comment]
I see, that makes sense. Glad this wasn't as difficult to figure out, by the sounds of it
